core/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -226,36 +251,16 @@ public Registration registerDispatchInterceptor(MessageDispatchInterceptor<Query
         return () -> dispatchInterceptors.remove(interceptor);
     }
 
-    private <Q, R> List<MessageHandler<? super QueryMessage<?, ?>>> getHandlersForMessage(QueryMessage<Q, R> queryMessage) {
-        return subscriptions.getOrDefault(new QueryDefinition(queryMessage.getQueryName(), queryMessage.getResponseType()),
-                                          new CopyOnWriteArrayList<>());
-    }
-
-    private static class QueryDefinition {
-        private final String queryName;
-        private final Class<?> responseType;
+    @SuppressWarnings("unchecked") // Suppresses 'queryHandler' cast to `MessageHandler<? super QueryMessage<?, ?>>`
+    private <Q, R> List<MessageHandler<? super QueryMessage<?, ?>>> getHandlersForMessage(
+            QueryMessage<Q, R> queryMessage) {
+        ResponseType<R> responseType = queryMessage.getResponseType();
 
-        private QueryDefinition(String queryName, Class<?> responseType) {
-            this.queryName = queryName;
-            this.responseType = responseType;
-        }
-
-        @Override
-        public boolean equals(Object o) {
-            if (this == o) {
-                return true;
-            }
-            if (!(o instanceof QueryDefinition)) {
-                return false;
-            }
-            QueryDefinition that = (QueryDefinition) o;
-            return Objects.equals(queryName, that.queryName) &&
-                    Objects.equals(responseType, that.responseType);
-        }
-
-        @Override
-        public int hashCode() {
-            return Objects.hash(queryName, responseType);
-        }
+        return subscriptions.getOrDefault(queryMessage.getQueryName(), new CopyOnWriteArrayList<>())
```

Comment:
try using getOrCompute instead. Besides delaying creation of the collection to the moment where a key actually wasn't found, it also stored the result in the map, making it somewhat faster for consecutive invocations.

Reply:
I assume you  mean `computeIfAbsent()`, but got suggestion. Adjusted it.